### PR TITLE
Refactor schema parser and taxonomy

### PR DESCRIPTION
**Pull request by tshemsedinov**

### CODE REVIEW

File: taxonomy/security/SystemUser.schema [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  Login: { domain: 'Login', required: true, unique: true },
+  Password: { domain: 'PasswordHash', required: true },
+  Groups: Many({ category: 'SystemGroup' }),
+  FullName: Include({ category: 'FullName' }),
+  Strengthn: Validate(record => {
+    if (record.Login.length < 6) return false;
+    if (record.Password..length < 6) return false;
```
Comment (lundibundi) [inline taxonomy/security/SystemUser.schema]: typo 2 dots.

File: taxonomy/security/SystemUser.schema [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    const val = record.MasterName.FieldName;
+    return record.FirldSetName.Field1.includes(val);
```
Comment (lundibundi) [inline taxonomy/security/SystemUser.schema]: `FirldSetName`?

File: taxonomy/security/SystemUser.schema [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  Login: { domain: 'Login', required: true, unique: true },
+  Password: { domain: 'PasswordHash', required: true },
+  Groups: Many({ category: 'SystemGroup' }),
+  FullName: Include({ category: 'FullName' }),
+  Strengthn: Validate(record => {
```
Comment (lundibundi) [inline taxonomy/security/SystemUser.schema]: Also a typo `Strengthn`

File: taxonomy/security/SystemUser.schema (hunk 2 above)
Comment (tshemsedinov) [reply taxonomy/security/SystemUser.schema]: That's just from example, I'll remove

**Comment by tshemsedinov** [comment]
Landed in 093a118 and 53881aa

**Pull request closed without merging**